fvm/transactionEnv.go

```diff
@@ -648,7 +643,15 @@ func (e *TransactionEnv) GetProgram(location common.Location) (*interpreter.Prog
 		}
 	}
 
-	program, has := e.programs.Get(location)
+	program, cached, has := e.programs.Get(location)
+
+	if !cached {
```

Comment:
This additional check is what we need to get the metered computation down to the 300 that the test expects. However, we need to also make sure to not double count memory metering as well as register touches; basically everything that gets merged. Is there a way to generalize this?

Reply:
Relatedly, what is the best way to access data like the number of interactions from within the tests?

Reply:
This is a bit controversial topic, I think the design is if we do touch anything we should measure it, and we might need to fix it by some cache on higher levels, if cadence runtime ask for it, we should measure it, and maybe we can do a cache on cadence runtime to not request for what has been loaded ?